dpcpp/src/CSWMixed.hpp

```diff
@@ -0,0 +1,1325 @@
+            q1(d.npx+1, d.npy,k) = q1(d.npx, d.npy-2,k);
+            q2(d.npx,   d.npy,k) = q2(d.npx, d.npy-1,k);
+            q2(d.npx+1, d.npy,k) = q2(d.npx, d.npy-2,k);
+        }
+        break;
+        case(2): // y-dir
+        if ( d.sw_corner )
+        {
+            q1(0,  0,k) = q1(1, 0,k);
+            q1(0, -1,k) = q1(2, 0,k);
+            q2(0,  0,k) = q2(1, 0,k);
+            q2(0, -1,k) = q2(2, 0,k);
+        }
+        if ( d.se_corner )
+        {
+            q1(d.npx,  0,k) = q1(d.npx-1, 0,k);
+            q1(d.npx, -1,k) = q1(d.npx-2, 0,k);
+            q2(d.npx,  0,k) = q2(d.npx-1, 0,k);
+            q2(d.npx, -1,k) = q2(d.npx-2, 0,k);
+        }
+        if ( d.nw_corner )
+        {
+            q1(0, d.npy,k)   = q1(1, d.npy,k);
+            q1(0, d.npy+1,k) = q1(2, d.npy,k);
+            q2(0, d.npy,k)   = q2(1, d.npy,k);
+            q2(0, d.npy+1,k) = q2(2, d.npy,k);
+        }
+        if ( d.ne_corner )
+        {
+            q1(d.npx, d.npy,k)   = q1(d.npx-1, d.npy,k);
+            q1(d.npx, d.npy+1,k) = q1(d.npx-2, d.npy,k);
+            q2(d.npx, d.npy,k)   = q2(d.npx-1, d.npy,k);
+            q2(d.npx, d.npy+1,k) = q2(d.npx-2, d.npy,k);
+        }
+        break;
+    }
+
+    #ifdef ENABLE_GPTL
+    if (do_profile == 1)
+    {
+        ret = gptlstop('fill2_4corners');
+    }
+    #endif
+
+    }
+
+    void d2a2c_vect(const CSWData& d, queue& Q)
+    {
+        const double zero=static_cast<double>(0.0);
+
+        #ifdef ENABLE_GPTL
+        int ret;
+        if (do_profile == 1)
+        {
+            ret = gptlstart ('d2a2c_vect');
+        }
+        #endif
+
+        //for (int k = 0; k < d.npz; k++) {
+        Q.parallel_for(range(d.npz, d.jsize, d.isize), [=](auto idx) {
+            // Initialize the non-existing corner regions
+            const int k = idx[0];
+            const int j = idx[1]+d.jsd;
+            const int i = idx[2]+d.isd;
+            //for (int j=d.jsd; j<=d.jed; j++)
+            //{
+            //    for (int i=d.isd; i<=d.ied; i++) 
+            //    {
+                    d.utmp(i,j,k)=d.big_number;
+                    d.vtmp(i,j,k)=d.big_number;
+            //    }
+            //}
+        }).wait();
+
+            //----------
+            // Interior:
+            //----------
+        Q.parallel_for(range(d.npz, std::min(d.npy-4,d.je+1)-std::max(4, d.js-1)+1, std::min(d.npx-4,d.ied)-std::max(4, d.isd)+1), [=](auto idx) {
+            const int k = idx[0];
+            const int j = idx[1]+std::max(4, d.js-1);
+            const int i = idx[2]+std::max(4, d.isd);
+            //for (int j = std::max(4, d.js-1); j<=std::min(d.npy-4,d.je+1); j++)
+            //{
+            //    for (int i = std::max(4, d.isd); i<=std::min(d.npx-4,d.ied); i++)
+            //    {
+                    d.utmp(i, j,k) = d.a2 * (d.u(i, j-1,k) + d.u(i, j+2,k)) + d.a1 * (d.u(i, j,k) + d.u(i, j+1,k));
+            //    }
+            //}
+        });
```

Comment:
Good question...  Initially I thought it was because nothing depended on this value in the following code.  Except, the vtmp variable below has a wait.  They both cannot be correct.  Both utmp and vtmp are used at line 480 so my guess this should have a wait.